openssl-dynamic/src/main/c/tcn_lock_rw.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef TCN_LOCK_RW_H
+#define TCN_LOCK_RW_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+typedef void* tcn_lock_rw_t;
```

Comment:
we have `lock_rw`, `lock_reader`, and `lock_writer`. Consider being consistent/complete with the naming:

- `lock_reader_writer`, `lock_reader`, and `lock_writer`
- `lock_rw`, `lock_r`, and `lock_w`

Also consider making the file name consistent if you go with the more verbose names.